argocd-cli.yaml

```diff
@@ -0,0 +1,45 @@
+package:
+  name: argocd-cli
+  version: 2.6.2
+  epoch: 0
+  description: "Declarative continuous deployment for Kubernetes."
+  target-architecture:
+    - all
+  copyright:
+    - paths:
+        - "*"
+      license: Apache License 2.0
+  dependencies:
+    runtime:
+      - ca-certificates-bundle
+environment:
+  contents:
+    packages:
+      - ca-certificates-bundle
+      - busybox
+      - go
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/argoproj/argo-cd
+      branch: v${{package.version}}
+      destination: argo-cd
+  #TODO(kamesh) - simplify this using go/build pipeline
+  # but it requires few OOTB variables
+  - name: build argocd cli
+    runs: |
+      set +x
+      cd argo-cd
+      mkdir -p "${{targets.destdir}}/usr/bin"
+      build_date=$(date -u +'%Y-%m-%dT%H:%M:%SZ')
+      git_commit=$(git rev-parse HEAD)
+      git_tree_state=$(git status --porcelain)
+      kubectl_version=$(go list -m k8s.io/client-go | head -n 1 | rev | cut -d' ' -f1 | rev)
+      CGO_ENABLED=0 go build -v \
+        -trimpath -ldflags \
+        "-X github.com/argoproj/argo-cd/v2/common.version=${{package.version}} -X github.com/argoproj/argo-cd/v2/common.buildDate==${build_date} -X github.com/argoproj/argo-cd/v2/common.gitCommit=${git_commit} -X github.com/argoproj/argo-cd/v2/common.gitTreeState=${git_tree_state} -X github.com/argoproj/argo-cd/v2/common.kubectlVersion=${kubectl_version}" \
+        -o "${{targets.destdir}}/usr/bin/argocd" ./cmd
+  - name: ensure binary has right version info
+    runs: |
+       ${{targets.destdir}}/usr/bin/argocd --version | grep --client "argocd: v${{package.version}}"
```

Comment:
```suggestion
       ${{targets.destdir}}/usr/bin/argocd --version | grep "argocd: v${{package.version}}"
```

Reply:
@imjasonh  - preferred to use `--client` as w/o that it checks for server version as well and if kube is not running then it will result in exit code 1.